tests/framework/db/mysql/SchemaTest.php

```diff
@@ -104,21 +104,23 @@ public function testAlternativeDisplayOfDefaultCurrentTimestampInMariaDB()
          * returns in response to the query `SHOW FULL COLUMNS FROM ...`
          */
         $schema = new Schema();
+        $this->setInaccessibleProperty($schema, 'db', $this->getConnection());
+
         $column = $this->invokeMethod($schema, 'loadColumnSchema', [[
             'field' => 'emulated_MariaDB_field',
             'type' => 'timestamp',
             'collation' => NULL,
             'null' => 'NO',
             'key' => '',
-            'default' => 'current_timestamp()',
+            'default' => new Expression('current_timestamp()'),
             'extra' => '',
             'privileges' => 'select,insert,update,references',
             'comment' => '',
         ]]);
 
         $this->assertInstanceOf(ColumnSchema::className(), $column);
         $this->assertInstanceOf(Expression::className(), $column->defaultValue);
-        $this->assertEquals('CURRENT_TIMESTAMP', $column->defaultValue);
+        $this->assertSame('current_timestamp()', (string)$column->defaultValue);
```

Comment:
Is this truly BC?

Reply:
seems to depend on the mariadb version:

https://github.com/yiisoft/yii2/pull/19758/files#diff-fb0db7affe9f204d44d0e48ccc6a2dbe1aa903ad6e7ed844d26a109c84e3abbdL299-L300

> When displayed in the INFORMATION_SCHEMA.COLUMNS table, a default CURRENT TIMESTAMP is displayed
> as CURRENT_TIMESTAMP up until MariaDB 10.2.2, and as current_timestamp() from MariaDB 10.2.3.
>
> See details here: https://mariadb.com/kb/en/library/now/#description

When we differentiate between value and expression now the change seems correct to me. However a note in UPGRADE would be useful.